SalesforceFoundation/gem
# W-034493 Provide Functionality to Update existing EDA accounts when GEM is installed

# Critical Changes

# Changes
* In orgs that had EDA installed previously, several system fields on account may need to be updated to ensure correct functionality. We've provided a global method which starts an asynchronous process to fix the accounts which need those updated values.
# Issues Closed
Closes #195
# New Metadata

# Deleted Metadata

# Testing Notes
* Install EDA in an org, create Household and Administrative accounts, and a few accounts that are neither of those types to ensure nothing happens to them. 
* Install GEM (dev_org cci flow is fine here).
* Observe that `npe01__SYSTEM_AccountType__c` is null, `npe01__SYSTEMIsIndividual__c` is false and Type is null for all Accounts that existed prior to GEM being installed.
* Run `Post_Install_Script.queueAccountProcessor();` via anonymous apex.
* Observe updated values in the above fields

Another option for testing:
* create a GEM dev scratch org
* disable the triggers for EDA and NPSP
* add the system fields npe01__SYSTEM_AccountType__c, npe01__SYSTEMIsIndividual__c and also Type on the account Layout
* create accounts: Household and Administrative
* make sure that npe01__SYSTEM_AccountType__c  is empty, Type is empty, and npe01__SYSTEMIsIndividual__c is false for the existing accounts
* Run `Post_Install_Script.queueAccountProcessor();` via anonymous apex.
* Observe updated values in the above fields
